walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,70 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+    final long MINUTES_IN_A_DAY = 1440;
+    final int START_OF_DAY = TimeRange.START_OF_DAY;
+    final int END_OF_DAY = TimeRange.END_OF_DAY;
+    int endOfTimeRange = 0;
+    TimeRange previousTimeRange = TimeRange.fromStartEnd(0,0,true);
+    Collection<TimeRange> unavailableTimeRange = new ArrayList<>();
+    Collection<TimeRange> unavailableNoOverlap = new ArrayList<>();
+    Collection<TimeRange> returnCollection = new ArrayList<>();
+
+    long durationOfRequest = request.getDuration();
+    Collection<String> attendeesOfRequest = request.getAttendees();
+
+    if(attendeesOfRequest.isEmpty()){
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }else if(durationOfRequest > MINUTES_IN_A_DAY){
+      return Arrays.asList();
+    }else{
```

Comment:
since you have `return` statements in the if and else if statements above, you can reformat as follows. This will make easier to read :) 

```
if(attendeesOfRequest.isEmpty()){
       return Arrays.asList(TimeRange.WHOLE_DAY);
}
if(durationOfRequest > MINUTES_IN_A_DAY){
       return Arrays.asList();
}

for(String attendee : attendeesOfRequest) 
```